### ISSUE TITLE
Reading string options using C API cannot be done safely

### ISSUE THREAD

**Issue by cyderize** [ERGO-Code/HiGHS#1112]
Hi, I'm trying to use the C API to get the value of a string option.

The signature of `Highs_getStringOptionValue` expects a buffer in which the string `value` can be stored:
```c
HighsInt Highs_getStringOptionValue(const void* highs, const char* option, char* value);
```

However, there doesn't appear to be a way to get the required size of the buffer, and there doesn't appear to be a particular upper bound on the length of string options (e.g. the `solution_file` option seems like it could be any length). Gurobi uses a limit of `GRB_MAX_STRLEN` (512 characters) for its strings, so maybe something similar could be introduced here, or maybe there should be another function to get the length of a string option. Another option would be to make `Highs_getStringOptionValue` return the required buffer size if `value` is `NULL`, then you'd call it once to get the buffer size, then again with the correctly sized buffer.

The C# API, being built on top of the C API, seems to suffer from this problem, as it crashes when reading string options with values longer than the default `StringBuilder` buffer size:

```c#
var highs = new HighsLpSolver();
highs.setStringOptionValue("solution_file", "this is an extremely long string which would overrun a typical StringBuilder buffer");
Console.WriteLine("Wrote long string value option");
string value;
highs.getStringOptionValue("solution_file", out value); // Crashes
Console.WriteLine(string.Format("Read long string value option: {0}", value));
```

**Comment by mingodad** [comment]
See also https://github.com/ERGO-Code/HiGHS/issues/1024

**Comment by jajhall** [comment]
Thanks. This will be fixed in HiGHS v2.0, as it would break the C API, and is referenced in #1090

**Comment by cyderize** [comment]
Ah, somehow I messed that discussion/issue. Thanks!
